compose/const.py

```diff
@@ -24,56 +24,16 @@
 WINDOWS_LONGPATH_PREFIX = '\\\\?\\'
 
 COMPOSEFILE_V1 = ComposeVersion('1')
-COMPOSEFILE_V2_0 = ComposeVersion('2.0')
-COMPOSEFILE_V2_1 = ComposeVersion('2.1')
-COMPOSEFILE_V2_2 = ComposeVersion('2.2')
-COMPOSEFILE_V2_3 = ComposeVersion('2.3')
-COMPOSEFILE_V2_4 = ComposeVersion('2.4')
-
-COMPOSEFILE_V3_0 = ComposeVersion('3.0')
-COMPOSEFILE_V3_1 = ComposeVersion('3.1')
-COMPOSEFILE_V3_2 = ComposeVersion('3.2')
-COMPOSEFILE_V3_3 = ComposeVersion('3.3')
-COMPOSEFILE_V3_4 = ComposeVersion('3.4')
-COMPOSEFILE_V3_5 = ComposeVersion('3.5')
-COMPOSEFILE_V3_6 = ComposeVersion('3.6')
-COMPOSEFILE_V3_7 = ComposeVersion('3.7')
-COMPOSEFILE_V3_8 = ComposeVersion('3.8')
+COMPOSEFILE_V4 = ComposeVersion('4.0')
 
 # minimum DOCKER ENGINE API version needed to support
 # features for each compose schema version
 API_VERSIONS = {
     COMPOSEFILE_V1: '1.21',
-    COMPOSEFILE_V2_0: '1.22',
-    COMPOSEFILE_V2_1: '1.24',
-    COMPOSEFILE_V2_2: '1.25',
-    COMPOSEFILE_V2_3: '1.30',
-    COMPOSEFILE_V2_4: '1.35',
-    COMPOSEFILE_V3_0: '1.25',
-    COMPOSEFILE_V3_1: '1.25',
-    COMPOSEFILE_V3_2: '1.25',
-    COMPOSEFILE_V3_3: '1.30',
-    COMPOSEFILE_V3_4: '1.30',
-    COMPOSEFILE_V3_5: '1.30',
-    COMPOSEFILE_V3_6: '1.36',
-    COMPOSEFILE_V3_7: '1.38',
-    COMPOSEFILE_V3_8: '1.38',
+    COMPOSEFILE_V4: '1.38',
```

Comment:
This whole mechanism to define required API version based on requested schema version should be replaced by a check that a specific field in actual compose file can't be implemented by the actual engine API. But I assume this is a complex change and we should get this implemented in a further PR

Reply:
Yes, this will be the next step.